zookeeper-server/src/main/java/org/apache/zookeeper/client/ZKClientConfig.java

```diff
@@ -59,6 +59,7 @@
     public static final String SECURE_CLIENT = ZooKeeper.SECURE_CLIENT;
     public static final int CLIENT_MAX_PACKET_LENGTH_DEFAULT = 4096 * 1024; /* 4 MB */
     public static final String ZOOKEEPER_REQUEST_TIMEOUT = "zookeeper.request.timeout";
+    public static final String ZOOKEEPER_SERVER_PRINCIPAL = "zookeeper.server.principal";
```

Comment:
Please document this new property in zookeeperProgrammers.md in "Client Configuration Parameters" section

Reply:
Thanks @arshadmohammad for pointing it out. Have added the corresponding documentation, as requested :)